datafusion/optimizer/src/simplify_predicates.rs

```diff
@@ -0,0 +1,194 @@
+    let mut less_predicates = Vec::new(); // Combines < and <=
+    let mut eq_predicates = Vec::new();
+
+    for pred in predicates {
+        match &pred {
+            Expr::BinaryExpr(BinaryExpr { left: _, op, right }) => {
+                let right_is_literal = right.is_literal();
+                match (op, right_is_literal) {
+                    (Operator::Gt, true)
+                    | (Operator::Lt, false)
+                    | (Operator::GtEq, true)
+                    | (Operator::LtEq, false) => greater_predicates.push(pred),
+                    (Operator::Lt, true)
+                    | (Operator::Gt, false)
+                    | (Operator::LtEq, true)
+                    | (Operator::GtEq, false) => less_predicates.push(pred),
+                    (Operator::Eq, _) => eq_predicates.push(pred),
+                    _ => unreachable!("Unexpected operator: {}", op),
+                }
+            }
+            _ => unreachable!("Unexpected predicate {}", pred.to_string()),
+        }
+    }
+
+    let mut result = Vec::new();
+
+    // If we have equality predicates, they're the most restrictive
+    if !eq_predicates.is_empty() {
+        if eq_predicates.len() > 1 {
+            result.push(Expr::Literal(ScalarValue::Boolean(Some(false)), None));
+        } else {
+            result.push(eq_predicates[0].clone());
+        }
+    } else {
+        // Handle all greater-than-style predicates (keep the most restrictive - highest value)
+        if !greater_predicates.is_empty() {
+            if let Some(most_restrictive) =
+                find_most_restrictive_predicate(&greater_predicates, true)?
+            {
+                result.push(most_restrictive);
+            } else {
+                result.extend(greater_predicates);
+            }
+        }
+
+        // Handle all less-than-style predicates (keep the most restrictive - lowest value)
+        if !less_predicates.is_empty() {
+            if let Some(most_restrictive) =
+                find_most_restrictive_predicate(&less_predicates, false)?
+            {
+                result.push(most_restrictive);
+            } else {
+                result.extend(less_predicates);
+            }
+        }
+    }
+
+    Ok(result)
+}
+
+fn find_most_restrictive_predicate(
+    predicates: &[Expr],
+    find_greater: bool,
+) -> Result<Option<Expr>> {
+    if predicates.is_empty() {
+        return Ok(None);
+    }
+
+    let mut most_restrictive = predicates[0].clone();
+    let mut best_value: Option<ScalarValue> = None;
+
+    for pred in predicates {
+        if let Expr::BinaryExpr(BinaryExpr { left, op: _, right }) = pred {
+            // Extract the literal value based on which side has it
+            let mut scalar_value = None;
+            if right.is_literal() {
+                if let Expr::Literal(scalar, None) = right.as_ref() {
+                    scalar_value = Some(scalar.clone());
+                }
```

Comment:
I think you could avoid clones here with something like

```suggestion
            if let Expr::Literal(scalar, None) = right {
                    scalar_value = Some(scalar);
             }
```

(and similarly for the other match statements)